pkg/scheduler/internal/cache/cache_test.go

```diff
@@ -1081,48 +1081,59 @@ func buildNodeInfo(node *v1.Node, pods []*v1.Pod) *framework.NodeInfo {
 // and remove.
 func TestNodeOperators(t *testing.T) {
 	// Test data
-	nodeName := "test-node"
-	cpu1 := resource.MustParse("1000m")
-	mem100m := resource.MustParse("100m")
 	cpuHalf := resource.MustParse("500m")
 	mem50m := resource.MustParse("50m")
 	resourceFooName := "example.com/foo"
-	resourceFoo := resource.MustParse("1")
+	resourceList1 := map[v1.ResourceName]string{
+		v1.ResourceCPU:                   "1000m",
+		v1.ResourceMemory:                "100m",
+		v1.ResourceName(resourceFooName): "1",
+	}
+	resourceList2 := map[v1.ResourceName]string{
+		v1.ResourceCPU:                   "500m",
+		v1.ResourceMemory:                "50m",
+		v1.ResourceName(resourceFooName): "2",
+	}
 
+	taints := []v1.Taint{
+		{
+			Key:    "test-key",
+			Value:  "test-value",
+			Effect: v1.TaintEffectPreferNoSchedule,
+		},
+	}
+	fake80Image := "gcr.io/80:latest"
+	fake300Image := "gcr.io/300:latest"
+	imageStatus1 := map[string]int64{
+		fake80Image:     int64(80 * mb),
+		"gcr.io/80:v1":  int64(80 * mb),
+		fake300Image:    int64(300 * mb),
+		"gcr.io/300:v1": int64(300 * mb),
+	}
+	imageStatus2 := map[string]int64{
+		"gcr.io/600:latest": int64(600 * mb),
+		fake80Image:         int64(80 * mb),
+		"gcr.io/900:latest": int64(900 * mb),
+	}
+	node1 := st.MakeNode().Name("test-node-1").Capacity(resourceList1).Taints(taints).Images(imageStatus1)
+	node2 := st.MakeNode().Name("test-node-2").Capacity(resourceList2).Taints(taints).Images(imageStatus2)
+	node3 := st.MakeNode().Name("test-node-3").Capacity(resourceList1).Taints(taints).Images(imageStatus1)
+	node4 := st.MakeNode().Name("test-node-4").Capacity(resourceList2).Taints(taints).Images(imageStatus2)
 	tests := []struct {
-		node *v1.Node
-		pods []*v1.Pod
+		nodes map[string]*v1.Node
+		pods  []*v1.Pod
 	}{
 		{
-			node: &v1.Node{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: nodeName,
-				},
-				Status: v1.NodeStatus{
-					Allocatable: v1.ResourceList{
-						v1.ResourceCPU:                   cpu1,
-						v1.ResourceMemory:                mem100m,
-						v1.ResourceName(resourceFooName): resourceFoo,
-					},
-				},
-				Spec: v1.NodeSpec{
-					Taints: []v1.Taint{
-						{
-							Key:    "test-key",
-							Value:  "test-value",
-							Effect: v1.TaintEffectPreferNoSchedule,
-						},
-					},
-				},
-			},
+			nodes: map[string]*v1.Node{node1.GetName(): &node1.Node, node2.GetName(): &node2.Node,
```

Comment:
you can be more explicit:

```golang
nodes: []*v1.Node{
  st.MakeNode()...,
  st.MakeNode()...
}
```

This avoids having to jump from one line to another when reading tests https://enterprisecraftsmanship.com/posts/dry-damp-unit-tests/